### PR TITLE
Test on python 3.12

### PR DESCRIPTION
**Pull request by akrherz**
- Use `setup-python` GHA to provide python
- Address `utcnow()` deprecation in #175

### CODE REVIEW

**Review by phobson — APPROVED** [review]

**Comment by phobson** [comment]
Thanks for pulling this together. Holding off merging since you didn't actually as for a review yet.

**Comment by akrherz** [comment]
@phobson Thanks.  I didn't want to tag you so close to a holiday.

**Comment by jdkloe** [comment]
Also from my side thanks for your fast response.
I can confirm that these changes fix the issue in the Fedora configuration, and it now again builds and tests just fine.
See: https://koji.fedoraproject.org/koji/taskinfo?taskID=102904082
So once this is merged #175 can be closed I think.

**Pull request merged**
